utils/log_parser.py

```diff
@@ -15,136 +15,214 @@
 
 import argparse
 import re
+from typing import (
+    List,
+    Dict,
+    Tuple,
+)
 
 # Maintaining all nodes
-NODES_MAP = {}
-
-# Maintaining all edges
-EDGES_MAP = {}
+NODES_MAP: Dict[Tuple[str, str], 'Node'] = {}
```

Comment:
Consider creating a `NamedTuple` for the key type of this map so that you can make it clear what each tuple component represents:

```
class NodeNameAndKind(typing.NamedTuple):
  name: str
  kind: str
```

This allows you to do something like this:
```
a = NodeNameAndKind("node", "kind")
print(a.name)
print(a.kind)
```